[
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "<@U1WUAQJVC>: from what I can see there aren't really any table queries... tables just return all their rows",
        "ts": "1471247781.001444"
    },
    {
        "user": "U21DSN0CX",
        "text": "<@U21DSN0CX|nickgogerty> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1471269769.001448"
    },
    {
        "user": "U21ECMHL4",
        "text": "<@U21ECMHL4|lefkok> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1471274521.001449"
    },
    {
        "user": "U21GVL6N6",
        "text": "<@U21GVL6N6|calvin.powers> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1471284106.001450"
    },
    {
        "type": "message",
        "subtype": "file_share",
        "text": "<@U0STZRN0Y|vijayr24> uploaded a file: <https:\/\/hyperledgerproject.slack.com\/files\/vijayr24\/F21HUGVRA\/snip20160815_1.png|Snip20160815_1.png> and commented: Hi All, I am trying to follow the steps mentioned in the following link \"<https:\/\/github.com\/hyperledger\/fabric\/blob\/master\/docs\/Setup\/Chaincode-setup.md#chaincode-deploy-via-cli-and-rest>\" to setup Hyperledger in my MAC using docker. I am getting the following error when I try to run my peer",
        "file": {
            "id": "F21HUGVRA",
            "created": 1471292094,
            "timestamp": 1471292094,
            "name": "Snip20160815_1.png",
            "title": "Snip20160815_1.png",
            "mimetype": "image\/png",
            "filetype": "png",
            "pretty_type": "PNG",
            "user": "U0STZRN0Y",
            "editable": false,
            "size": 88197,
            "mode": "hosted",
            "is_external": false,
            "external_type": "",
            "is_public": true,
            "public_url_shared": false,
            "display_as_bot": false,
            "username": "",
            "url_private": "https:\/\/files.slack.com\/files-pri\/T0J024XGA-F21HUGVRA\/snip20160815_1.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "url_private_download": "https:\/\/files.slack.com\/files-pri\/T0J024XGA-F21HUGVRA\/download\/snip20160815_1.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_64": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_64.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_80": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_80.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_360": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_360.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_360_w": 360,
            "thumb_360_h": 37,
            "thumb_480": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_480.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_480_w": 480,
            "thumb_480_h": 50,
            "thumb_160": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_160.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_720": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_720.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_720_w": 720,
            "thumb_720_h": 75,
            "thumb_960": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_960.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_960_w": 960,
            "thumb_960_h": 100,
            "thumb_1024": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F21HUGVRA-3578f64d19\/snip20160815_1_1024.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_1024_w": 1024,
            "thumb_1024_h": 107,
            "image_exif_rotation": 1,
            "original_w": 1844,
            "original_h": 192,
            "permalink": "https:\/\/hyperledgerproject.slack.com\/files\/vijayr24\/F21HUGVRA\/snip20160815_1.png",
            "permalink_public": "https:\/\/slack-files.com\/T0J024XGA-F21HUGVRA-2693e8be3b",
            "channels": [
                "C0J02NU1E"
            ],
            "groups": [],
            "ims": [],
            "comments_count": 1,
            "initial_comment": {
                "id": "Fc21HUH8Q4",
                "created": 1471292094,
                "timestamp": 1471292094,
                "user": "U0STZRN0Y",
                "is_intro": true,
                "comment": "Hi All, I am trying to follow the steps mentioned in the following link \"<https:\/\/github.com\/hyperledger\/fabric\/blob\/master\/docs\/Setup\/Chaincode-setup.md#chaincode-deploy-via-cli-and-rest>\" to setup Hyperledger in my MAC using docker. I am getting the following error when I try to run my peer",
                "channel": ""
            }
        },
        "user": "U0STZRN0Y",
        "upload": true,
        "display_as_bot": false,
        "username": "<@U0STZRN0Y|vijayr24>",
        "bot_id": null,
        "ts": "1471292097.001451"
    },
    {
        "type": "message",
        "user": "U0STZRN0Y",
        "text": "When I execute the step \"CORE_CHAINCODE_ID_NAME=mycc CORE_PEER_ADDRESS=0.0.0.0:30303 .\/chaincode_example02\"",
        "ts": "1471292154.001452"
    },
    {
        "type": "message",
        "user": "U0STZRN0Y",
        "text": "I am using option 2 to setup Hyperledger",
        "ts": "1471292167.001453"
    },
    {
        "type": "message",
        "user": "U1XTQ3RS6",
        "text": "zmanage",
        "ts": "1471292492.001454"
    },
    {
        "user": "U21J007ML",
        "text": "<@U21J007ML|arnofontaine> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1471292656.001455"
    },
    {
        "user": "U21HV589Z",
        "text": "<@U21HV589Z|bgattas> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1471300416.001456"
    },
    {
        "type": "message",
        "user": "U20CGK1JQ",
        "text": "<@U0STZRN0Y>: the port changed from 30303.  Try port 7051",
        "ts": "1471309447.001457"
    },
    {
        "type": "message",
        "user": "U20CGK1JQ",
        "text": "I suggest that we either get the online docs updated or put up a notice that the default port number changed as there are multiple queries coming through here?",
        "ts": "1471309656.001458"
    },
    {
        "type": "message",
        "user": "U0TFEHX8E",
        "text": "which docs are you looking at? I believe most were updated with the change",
        "ts": "1471311979.001459"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "<@U0STZRN0Y>: if you are going to use CORE_PEER_ADDRESS=0.0.0.0:30303 you will need to set CORE_PEER_ADDRESSAUTODETECT=true\n\notherwise you need to set CORE_PEER_ADDRESS to a valid hostname\/IP that other machines can connect to\n\n<@U0TFEHX8E> see the github link vijayr24 posted",
        "ts": "1471312031.001460"
    },
    {
        "type": "message",
        "user": "U0TFEHX8E",
        "text": "ah, everything has been moved to Gerrit. <https:\/\/gerrit.hyperledger.org>\nThe GitHub docs are outdated at this point",
        "ts": "1471312088.001461"
    },
    {
        "type": "message",
        "user": "U11HH3P7Y",
        "text": "<@U1LES78TT>: Please refer this url <http:\/\/169.53.62.117\/site\/index.html>",
        "attachments": [
            {
                "title": "Hyperledger Fabric",
                "title_link": "http:\/\/169.53.62.117\/site\/index.html",
                "text": "Welcome to the Hyperledger fabric documentation",
                "fallback": "Hyperledger Fabric",
                "from_url": "http:\/\/169.53.62.117\/site\/index.html",
                "service_name": "169.53.62.117",
                "id": 1
            }
        ],
        "ts": "1471312122.001462"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "<@U11HH3P7Y>: thanks, why am I referring to that URL?",
        "ts": "1471312155.001464"
    },
    {
        "type": "message",
        "user": "U0TFEHX8E",
        "text": "I realize this is a huge point of confusion currently. I hope that Gerrit will be mirrored in GitHub soon. Maybe <@U0UHG4VP1> or <@U0KM61BCP> know the status",
        "edited": {
            "user": "U0TFEHX8E",
            "ts": "1471312177.000000"
        },
        "ts": "1471312164.001465"
    },
    {
        "type": "message",
        "user": "U0TFEHX8E",
        "text": "found the issue tracking the mirroring <https:\/\/jira.hyperledger.org\/browse\/FAB-77>",
        "ts": "1471312233.001467"
    },
    {
        "type": "message",
        "user": "U0KM61BCP",
        "text": "been traveling today, my plan is to move the archives and get the mirror in place this week",
        "ts": "1471312268.001468",
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "U0TFEHX8E",
                    "U11HH3P7Y"
                ],
                "count": 2
            }
        ]
    },
    {
        "type": "message",
        "user": "U0TFEHX8E",
        "text": "also there\u2019s a mailing list where these types of updates are posted. <http:\/\/lists.hyperledger.org\/pipermail\/hyperledger-fabric\/2016-August\/000189.html>\nYou can see all the lists and subscribe at <http:\/\/lists.hyperledger.org\/mailman\/listinfo>",
        "ts": "1471312398.001469"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "<@U0KM61BCP> is <http:\/\/hyperledger.org\/> going to be updated as well at some point?",
        "ts": "1471312441.001470"
    },
    {
        "type": "message",
        "user": "U0KM61BCP",
        "text": "yes",
        "ts": "1471312454.001471"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "great :slightly_smiling_face:",
        "ts": "1471312472.001472"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "Anyone deployed car-lease-demo to local fabric instance?",
        "ts": "1471313429.001473"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "I am having trouble deploying it. getting following errors:\n```02:06:56.673 [crypto] getEnrollmentCertificateFromECA -&gt; ERRO 01c [client.DVLA] Failed invoking CreateCertficatePair [rpc error: code = 2 desc = \"Identity lookup error: sql: no rows in result set\"].\n02:06:56.674 [crypto] retrieveEnrollmentData -&gt; ERRO 01d [client.DVLA] Failed getting enrollment certificate [id=DVLA]: [rpc error: code = 2 desc = \"Identity lookup error: sql: no rows in result set\"]\n02:06:56.674 [crypto] registerCryptoEngine -&gt; ERRO 01e [client.DVLA] Failed retrieving enrollment data [rpc error: code = 2 desc = \"Identity lookup error: sql: no rows in result set\"].\n02:06:56.674 [crypto] register -&gt; ERRO 01f [client.DVLA] Failed registering node crypto engine [rpc error: code = 2 desc = \"Identity lookup error: sql: no rows in result set\"].\n02:06:56.674 [crypto] Login -&gt; ERRO 020 Failed registering client [DVLA] with name [DVLA] [rpc error: code = 2 desc = \"Identity lookup error: sql: no rows in result set\"].\n02:06:56.674 [rest] Register -&gt; ERRO 021 Error on client login: rpc error: code = 2 desc = 'Identity lookup error: sql: no rows in result set'\n02:06:56.771 [rest] Register -&gt; INFO 022 REST client login...\n02:06:56.771 [rest] Register -&gt; ERRO 023 {\"Error\": \"enrollId and enrollSecret may not be blank.\"}\n```",
        "edited": {
            "user": "U1Y2R5C9L",
            "ts": "1471313661.000000"
        },
        "ts": "1471313500.001474"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "this is happening for all users on startup",
        "ts": "1471313519.001475"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "Any help would be appreciated",
        "ts": "1471313594.001477"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "<@U11MANG9G>: may be you can help?",
        "ts": "1471317910.001479"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Hi <@U1Y2R5C9L>!",
        "ts": "1471318061.001480"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "So the last error is thrown at the REST API layer right after marshalling the login spec.",
        "ts": "1471318097.001481"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "ok",
        "ts": "1471318120.001482"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Seems like the lookup for the EnrollmentID could not be found.",
        "ts": "1471318135.001483"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I need some more info as for how you run it...",
        "ts": "1471318149.001484"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "sure",
        "ts": "1471318165.001485"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "(and to be clear, these error messages are not clear!)",
        "ts": "1471318167.001486"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "Agreed",
        "ts": "1471318177.001487"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "(don't think these have gone unnoticed)",
        "ts": "1471318184.001488"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "I am starting membership service first and then the peer node in a development environment",
        "ts": "1471318220.001489"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "then I start the car-lease-demo app using node server",
        "ts": "1471318268.001490"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "the demo tries to create the users",
        "ts": "1471318288.001491"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "this is when these errors appear",
        "ts": "1471318300.001492"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "I deployed this to Blumix, and it worked without any issue",
        "ts": "1471318324.001493"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "So I guess there is some specific setup configuration that I am missing",
        "ts": "1471318351.001494"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Yes, that's what I think too.",
        "ts": "1471318369.001495"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I can't see the setup - I understand, but I can try to give you more info.",
        "ts": "1471318385.001496"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "sure",
        "ts": "1471318395.001497"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "The top exception (before the unwinding)... starts at the node...",
        "ts": "1471318414.001498"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "ok",
        "ts": "1471318438.001499"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "The EnrollID is 'DVLA' and the node is trying to get an ECert from the ECA with a username password pair",
        "ts": "1471318449.001500"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "ok",
        "ts": "1471318461.001501"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I don't know what your configuration is...",
        "ts": "1471318461.001502"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "But the ECA is trying to retrieve a record from its local db, and I believe that this is where everything is breaking...",
        "ts": "1471318487.001503"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "... there isn't a record for DVLA (I suspect)",
        "ts": "1471318506.001504"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "I am using instructions from here ```<https:\/\/github.com\/IBM-Blockchain\/car-lease-demo\/blob\/master\/Documentation\/Installation%20Guide.md#deploying-locally111>",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "IBM-Blockchain\/car-lease-demo",
                "title_link": "https:\/\/github.com\/IBM-Blockchain\/car-lease-demo\/blob\/master\/Documentation\/Installation%20Guide.md#deploying-locally111",
                "text": "car-lease-demo - A demonstration using IBM Blockchain to show how the lifecycle of vehicles can be recorded on a blockchain",
                "fallback": "GitHub: IBM-Blockchain\/car-lease-demo",
                "thumb_url": "https:\/\/avatars2.githubusercontent.com\/u\/17178768?v=3&s=400",
                "from_url": "https:\/\/github.com\/IBM-Blockchain\/car-lease-demo\/blob\/master\/Documentation\/Installation%20Guide.md#deploying-locally111",
                "thumb_width": 200,
                "thumb_height": 200,
                "service_icon": "https:\/\/github.com\/apple-touch-icon.png",
                "id": 1
            }
        ],
        "ts": "1471318543.001505"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "I agree, but demo should be creating that record",
        "ts": "1471318570.001507"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "So I believe that it's to do with the ECA's configuration...",
        "ts": "1471318573.001508"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Yes, the demo should be creating that record that you are asking for.",
        "ts": "1471318590.001509"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "ECA is the membersrvc right?",
        "ts": "1471318620.001510"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Yes, the ECA's (and others') configuration are in fabric\/membersrvc\/membersrvc.yaml",
        "ts": "1471318661.001511"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "` fabric\/membersrvc\/membersrvc.yaml` for the artists among us (== nicer colors)",
        "edited": {
            "user": "U11MANG9G",
            "ts": "1471318686.000000"
        },
        "ts": "1471318679.001512"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "There are both initial users\/IDs that are pre-configured, and also the ability to register new ones dynamically",
        "ts": "1471318717.001514"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "yes I see that, but DVLA is not one of them",
        "ts": "1471318783.001515"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "actually DVLA is in the demo's membersrvc.yaml",
        "ts": "1471318872.001516"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "let me try with this one",
        "ts": "1471318893.001517"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "yes",
        "ts": "1471319150.001519"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I think we should also slightly change these errors (especially in dev mode)",
        "ts": "1471319154.001521"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "[Sorry, a typo]",
        "ts": "1471319163.001522"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "To assist us with understanding what's going on.",
        "ts": "1471319194.001523"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "BUT the ECA and the TCA are undergoing some more changes... that should make such things easier.",
        "ts": "1471319234.001524"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I'm wondering what other REST (API) calls were expected... after the ECA step. But one thing at a time.",
        "ts": "1471319290.001525"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "How does it look over there? :wink:",
        "ts": "1471319295.001526"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "It is looking good so far",
        "ts": "1471319308.001527"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "no more errors",
        "ts": "1471319327.001528"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "somehow I messed up membersrvc.yaml.demo and the actual membersrvc.yaml",
        "ts": "1471319352.001529"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "not sure if ```03:48:36.439 [dockercontroller] Start -&gt; ERRO 0fa start-could not start container API error (400): {\"message\":\"starting container with HostConfig was deprecated since v1.10 and removed in v1.12\"}``` is really an error",
        "ts": "1471319391.001530"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "tried to create scenarios using the demo but it failed with a lot of these errors",
        "ts": "1471319487.001531"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "perhaps I should try with the latest code base",
        "ts": "1471319497.001532"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "Thanks <@U11MANG9G> I'll try this with HEAD and see if I can fix these error",
        "ts": "1471319687.001533"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "will ping you again if I have anything to ask. Thanks for your help",
        "ts": "1471319707.001534"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Good stuff. Yes, feel free to reach out.",
        "ts": "1471319721.001535"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I'm usually monitoring the Membership Services channel more closely... but if you @ ... me then I'd jump on it.",
        "ts": "1471319763.001536"
    },
    {
        "type": "message",
        "user": "U1Y2R5C9L",
        "text": "sure, thanks",
        "ts": "1471320255.001537"
    },
    {
        "type": "message",
        "user": "U1WUAQJVC",
        "text": "<@U1LES78TT>:  I also found the same. Hence wanted to confirm. Thank you.",
        "ts": "1471321222.001538"
    }
]